Networking/Networking/Remote/ProductCategoriesRemote.swift

```diff
@@ -30,6 +30,33 @@ public final class ProductCategoriesRemote: Remote {
 
         enqueue(request, mapper: mapper, completion: completion)
     }
+
+    /// Create a new `ProductCategory`.
+    ///
+    /// - Parameters:
+    ///     - siteID: Site for which we'll add a new product categories.
+    ///     - name: Category name.
+    ///     - parentID: The ID for the parent of the category.
+    ///     - completion: Closure to be executed upon completion.
+    ///
+    public func createProductCategory(for siteID: Int64,
+                                      name: String,
+                                      parentID: Int64?,
+                                      completion: @escaping (ProductCategory?, Error?) -> Void) {
```

Comment:
⚠️ We should use `Result` now that it's available. 🙂 We added it in https://github.com/woocommerce/woocommerce-ios/pull/2261. 

```suggestion
    public func createProductCategory(for siteID: Int64,
                                      name: String,
                                      parentID: Int64?,
                                      completion: @escaping (Result<ProductCategory, Error>) -> Void) {
```